test/functional/wallet_inactive_hdchains.py

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env python3
+# Copyright (c) 2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""
+Test Inactive HD Chains.
+"""
+import os
+import shutil
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.wallet_util import (
+    get_generate_key,
+)
+
+
+class InactiveHDChainsTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+        self.extra_args = [["-keypool=10"], ["-nowallet", "-keypool=10"]]
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+        self.skip_if_no_bdb()
+        self.skip_if_no_previous_releases()
+
+    def setup_nodes(self):
+        self.add_nodes(self.num_nodes, extra_args=self.extra_args, versions=[
+            None,
+            170200, # 0.17.2 Does not have the key metadata upgrade
+        ])
+
+        self.start_nodes()
+        self.init_wallet(node=0)
+
+    def test_basic(self):
+        self.log.info("Test basic case for inactive HD chains")
+        default = self.nodes[0].get_wallet_rpc(self.default_wallet_name)
+
+        self.nodes[0].createwallet(wallet_name="basic_base", descriptors=False, blank=True)
+        self.nodes[0].createwallet(wallet_name="basic_test", descriptors=False, blank=True)
+        base_wallet = self.nodes[0].get_wallet_rpc("basic_base")
+        test_wallet = self.nodes[0].get_wallet_rpc("basic_test")
+
+        # Setup both wallets with the same HD seed
+        seed = get_generate_key()
+        base_wallet.sethdseed(True, seed.privkey)
+        test_wallet.sethdseed(True, seed.privkey)
+
+        # Generate a new HD seed on the test wallet
+        test_wallet.sethdseed()
+
+        # Get the address at index 6
```

Comment:
where are these magic numbers coming from?

Reply:
Not entirely sure, but probably something to do with the keypool defaults. However these don't make sense with the keypool options we are using in the test, so I've changed this to match those.